cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    candu.disable_all()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_custom_rate(new_compute_rate):
+    """Assign custom Compute rate to the Enterprise and then queue the Chargeback report."""
+    description = new_compute_rate
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute rate')
+
+    yield
+
+    # Resetting the Chargeback rate assignment
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': '<Nothing>'}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+
+
+def verify_records_rollups_table(appliance, provider):
+    """Verify that hourly rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        return any([record.cpu_usagemhz_rate_average,
+           record.cpu_usage_rate_average,
+           record.derived_memory_used,
+           record.net_usage_rate_average,
+           record.disk_usage_rate_average])
+    return False
+
+
+def verify_records_metrics_table(appliance, provider):
+    """Verify that rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    metrics = appliance.db.client['metrics']
+
+    # Capture real-time C&U data
+    ret = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_capture('realtime', 1.hour.ago.utc, Time.now.utc)\""
+        .format(provider.id, repr(vm_name)))
+    assert ret.success, "Failed to capture VM C&U data:".format(ret.output)
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(metrics.id)
+            .join(ems, metrics.parent_ems_id == ems.id)
+            .filter(metrics.capture_interval_name == 'realtime',
+            metrics.resource_name == vm_name,
+            ems.name == provider.name, metrics.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(metrics).filter(
```

Comment:
QUESTION: I see several very similar code blocks. can those be combined ?

Reply:
@izapolsk this is something I had discussed with @nachandr earlier, although these look like the blocks could be replaced, the way the logic is operating the queries have to be made separately.